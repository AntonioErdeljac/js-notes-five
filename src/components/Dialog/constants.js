export const TEST_ENV = 'test';
