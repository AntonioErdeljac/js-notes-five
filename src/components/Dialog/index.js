import Modal from 'react-modal';
import PropTypes from 'prop-types';
import React, { memo } from 'react';

Modal.setAppElement('#root');

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
};

const Dialog = ({ isOpen, onClose, children }) => {
  return (
    <Modal isOpen={isOpen} onRequestClose={onClose} style={customStyles}>
      {children}
    </Modal>
  );
};

Dialog.propTypes = {
  isOpen: PropTypes.bool,
  onClose: PropTypes.func,
  children: PropTypes.node.isRequired,
};

Dialog.defaultProps = {
  isOpen: false,
  onClose: null,
};

export default memo(Dialog);
