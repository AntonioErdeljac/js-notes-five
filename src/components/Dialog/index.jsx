import Modal from 'react-modal';
import PropTypes from 'prop-types';
import React, { memo } from 'react';

import './index.scss';

Modal.setAppElement('#root');

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    height: '90vh',
    width: '40vw',
    border: 'none',
    padding: 30,
    boxShadow: '0px 0px 10px rgba(0,0,0,0.07)',
  },
};

const Dialog = ({ isOpen, onClose, children }) => {
  return (
    <Modal isOpen={isOpen} onRequestClose={onClose} style={customStyles}>
      {children}
    </Modal>
  );
};

Dialog.propTypes = {
  isOpen: PropTypes.bool,
  onClose: PropTypes.func,
  children: PropTypes.node.isRequired,
};

Dialog.defaultProps = {
  isOpen: false,
  onClose: null,
};

export default memo(Dialog);
