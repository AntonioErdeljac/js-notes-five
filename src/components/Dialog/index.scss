.ReactModal__Content {
  @media only screen and (max-width: 992px) {
    height: 100% !important;
    width: 100% !important;
  }
}

.ReactModal__Overlay--after-open {
  background-color: rgba(0,0,0,0.2) !important;
}
