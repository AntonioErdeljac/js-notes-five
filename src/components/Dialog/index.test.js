import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Dialog from './index';

const makeProps = ({ isOpen, onClose, children } = {}) => ({
  isOpen: isOpen || (() => {}),
  onClose: onClose || (() => {}),
  children: children || 'Content',
});

const renderComponent = (props) => shallow(<Dialog {...props} />);

describe('Dialog Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent(makeProps()))).toMatchSnapshot();
  });

  it('renders correct children', () => {
    const expectedChildren = 'Content';

    const props = makeProps({ chidlren: expectedChildren });
    const wrapper = renderComponent(props);

    expect(wrapper.children().debug()).toEqual(expectedChildren);
  });
});
