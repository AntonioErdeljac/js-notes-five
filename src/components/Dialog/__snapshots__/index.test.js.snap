// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component renders correctly according to snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  onRequestClose={[Function]}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "border": "none",
        "bottom": "auto",
        "boxShadow": "0px 0px 10px rgba(0,0,0,0.07)",
        "height": "90vh",
        "left": "50%",
        "marginRight": "-50%",
        "padding": 30,
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "40vw",
      },
    }
  }
>
  Content
</Modal>
`;
